pennylane/transforms/dynamic_one_shot.py

```diff
@@ -195,26 +206,35 @@ def init_auxiliary_tape(circuit: qml.tape.QuantumScript):
                 new_measurements.append(SampleMP(obs=m.obs))
             else:
                 new_measurements.append(m)
-    for op in circuit:
-        if is_mcm(op):
+    new_operations = []
+    for op in circuit.operations:
+        if "MidCircuitMeasure" in str(type(op)):  # pragma: no cover
+            new_op = op
+            new_op.bypass_postselect = True
```

Comment:
I mentioned this on slack, but I don't think we should set this attribute in PennyLane to effect a behaviour in Catalyst, because it is rather random. For now, leaving it out entirely means we just always consider it False (i.e. using "fill-shots" mode), and when Mudit implements the config handling on the catalyst side the bypass can be effected there based on the config.

Reply:
Now that the "fill-shots" mode exists, I'm all for it.